src/main/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoProcessor.java

```diff
@@ -0,0 +1,430 @@
+                                if (match) {
+                                    ingestDocument.setFieldValue(targetField, geoDataList);
+                                }
+                                handler.accept(ingestDocument, null);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(final Exception e) {
+                            handler.accept(null, e);
+                        }
+                    }
+                );
+            }
+
+            @Override
+            public void onFailure(final Exception e) {
+                handler.accept(null, e);
+            }
+        });
+    }
+
+    private Map<String, Object> filteredGeoData(final Map<String, Object> geoData, final String ip) {
+        Map<String, Object> filteredGeoData;
+        if (properties == null) {
+            filteredGeoData = geoData;
+        } else {
+            filteredGeoData = new HashMap<>();
+            for (String property : this.properties) {
+                if (property.equals(PROPERTY_IP)) {
+                    filteredGeoData.put(PROPERTY_IP, ip);
+                } else {
+                    filteredGeoData.put(property, geoData.get(property));
+                }
+            }
+        }
+        return filteredGeoData;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Ip2Geo processor factory
+     */
+    public static final class Factory implements Processor.Factory {
+        private final Ip2GeoCache cache;
+        private final Client client;
+        private final IngestService ingestService;
+        private TimeValue timeout;
+
+        /**
+         * Default constructor
+         *
+         * @param cache the cache
+         * @param client the client
+         * @param ingestService the ingest service
+         */
+        public Factory(final Ip2GeoCache cache, final Client client, final IngestService ingestService) {
+            this.cache = cache;
+            this.client = client;
+            this.ingestService = ingestService;
+
+            timeout = Ip2GeoSettings.TIMEOUT_IN_SECONDS.get(client.settings());
+            ClusterSettings clusterSettings = ingestService.getClusterService().getClusterSettings();
+            clusterSettings.addSettingsUpdateConsumer(Ip2GeoSettings.TIMEOUT_IN_SECONDS, newValue -> timeout = newValue);
+        }
+
+        /**
+         * When a user create a processor, this method is called twice. Once to validate the new processor and another
+         * to apply cluster state change after the processor is added.
+         *
+         * The second call is made by ClusterApplierService. Therefore, we cannot access cluster state in the call.
+         * That means, we cannot even query an index inside the call.
+         *
+         * Because the processor is validated in the first call, we skip the validation in the second call.
```

Comment:
if don't skip the validation what will happen?

Reply:
It throws exception.